src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
`fail_fast` seems to be what i described as async_wait_and_get_response for the current situation that there are no non error async responses. I tend to prefer two distinct method names over a barrier boolean parameter. But that is of course a style detail. Also wait is my preferred color of the bikeshed.